demos/python_demos/face_recognition_demo.md

```diff
@@ -0,0 +1,179 @@
+# Interactive Face Recognition Demo
+
+This example demonstrates an approach to create interactive applications
+for video processing. It shows the basic architecture for building model
+pipelines supporting model placement on different devices and simultaneous
+parallel or sequential execution using OpenVINO library in Python.
+In particular, this demo uses 4 models to build a pipeline able to detect
+faces on videos, detect face direction and its keypoints (aka "landmarks"),
+and recognize persons using provided face database (gallery). The corresponding
+pretrained models can be bound in this repository:
+
+* `intel_models/face-detection-retail-0004`,
+    which is used to detect faces and predict their bounding boxes;
+* `intel_models/head-pose-estimation-adas-0001`,
+    which is used to predict head direction;
+* `intel_models/landmarks-regression-retail-0009`,
+    which is used to predict face keypoints;
+* `intel_models/face-reidentification-retail-0071`,
+    which is used to recognize persons.
+
+### How it works
+
+The application is invoked from command line. It reads the specified input
+video stream frame-by-frame, be it a camera device or a video file,
+and performs independent analysis of each frame. In order to make predictions
+the application deploys 4 models on the specified devices using OpenVINO
+library and runs them in asynchronous manner. The first model to be used
+is the face detection model, which is followed by two independent models
+for head pose prediction and face keypoints prediction. The last step in
+frame processing is done by face recognition model, which uses keypoints
+to align the faces and the face gallery to match faces found on a video
+frame with the ones in the gallery. Then, the processing results are
+visualized and displayed on the screen or written to the output file.
+
+### Creating a gallery for face recognition
+
+In order to recognize faces the application needs a face database, or a gallery.
+The gallery is a folder with images of persons. Each image in the gallery can
+be of arbitrary size and should contain a tight crop of face. To obtain better
+results use images of square shapes. Image file name is used as a person name
+during the visualization.
+Use the following name convention: `person_N_name.png` or `person_N_name.jpg`.
+
+### Installation and dependencies
+
+The demo depends on:
+- OpenVINO library (R4)
+- Python (any of 2.7+ or 3.5+, which is supported by OpenVINO)
+- NumPy
```

Comment:
@zhiltsov-max please specify particular versions of **NumPy**, **SciPy** and **OpenCV** and add them into `requirements.txt` file as well.